lib/rubocop/rspec/blank_line_separation.rb

```diff
@@ -25,6 +25,16 @@ def offending_loc(last_line)
         source_range(processed_source.buffer, last_line, start, content_length)
       end
 
+      def single_node?(node)
+        return if node.parent.nil?
+
+        if node.parent.if_type?
```

Comment:
Is there a reason you've chosen to solve for only the if type explicitly? I imagine there are other constructs that would have this same problem (my first thought is `case`, but i possibly also `while` or `for` loops or something i'm not thinking of)

Would @bquorning's thought about checking if it's in a block/begin node not work https://github.com/rubocop-hq/rubocop-rspec/issues/648#issuecomment-397804501 ?

Reply:
You are right, we have to check `begin` nodes. I even don't see why a block node have to be checked - if there is more then one child of a block, it have to be wrapped in a begin.

Also, it's not single_node, but last_child

PR updated